src/ray/core_worker/core_worker.cc

```diff
@@ -89,22 +89,39 @@ class ScopedTaskMetricSetter {
 using ActorLifetime = ray::rpc::JobConfig_ActorLifetime;
 
 // Helper function converts GetObjectLocationsOwnerReply to ObjectLocation
-ObjectLocation CreateObjectLocation(
+std::shared_ptr<ObjectLocation> CreateObjectLocation(
     const rpc::WorkerObjectLocationsPubMessage &object_info) {
   std::vector<NodeID> node_ids;
   node_ids.reserve(object_info.node_ids_size());
   for (int i = 0; i < object_info.node_ids_size(); ++i) {
     node_ids.push_back(NodeID::FromBinary(object_info.node_ids(i)));
   }
   bool is_spilled = !object_info.spilled_url().empty();
-  return ObjectLocation(NodeID::FromBinary(object_info.primary_node_id()),
-                        object_info.object_size(),
-                        std::move(node_ids),
-                        is_spilled,
-                        object_info.spilled_url(),
-                        NodeID::FromBinary(object_info.spilled_node_id()),
-                        object_info.did_spill());
+  return std::make_shared<ObjectLocation>(
+      NodeID::FromBinary(object_info.primary_node_id()),
+      object_info.object_size(),
+      std::move(node_ids),
+      is_spilled,
+      object_info.spilled_url(),
+      NodeID::FromBinary(object_info.spilled_node_id()),
+      object_info.did_spill());
 }
+
+std::shared_ptr<ObjectLocation> TryGetLocalObjectLocation(
+    ReferenceCounter &reference_counter, const ObjectID &object_id) {
+  if (!reference_counter.HasReference(object_id)) {
+    return nullptr;
+  }
+  rpc::WorkerObjectLocationsPubMessage object_info;
+  reference_counter.FillObjectInformation(object_id, &object_info);
+  // Note: there can be a TOCTOU race condition: HasReference returned true, but before
+  // FillObjectInformation the object is released. Hence we check the ref_removed field.
+  if (object_info.ref_removed()) {
+    return nullptr;
+  }
+  return CreateObjectLocation(object_info);
```

Comment:
no because we want to return "Object Location" which we already know partially. But the object_size can be set to None in python. Fixed in CObjectLocationPtrToDict.

Reply:
done